Automattic/wp-calypso
# Reader: Update feed discovery to use data getters and data layer

Feed discovery was a feature I worked on during my trial almost 4 years ago, and I can see that back then I introduced some mess. This PR is cleaning that up.

So far, feed discovery was relying on the `lib/inflight` library (and was the last user of that), and it unnecessarily used LRU cache. 

This PR updates the feed discovery handler to work with data getters that use the data layer, and removes the no longer necessary `lib/feed-lookup`, `lib/inflight` and a corresponding method from `wpcom.undocumented()`.

#### Changes proposed in this Pull Request

* State: Add a `requestFeedDiscovery` data getter
* Reader: Use `requestFeedDiscovery` data getter for feed discovery
* Reader: Fix an incorrect request to `/read/feeds/NaN` when using feed discovery with an unexisting blog.
* Remove obsolete `lib/feed-lookup` library.
* Remove obsolete `lib/inflight` library.
* Remove the legacy `discoverFeed` method from `wpcom.undocumented()`.

#### Testing instructions

* Checkout this branch and build it.
* Go to http://calypso.localhost:3000/read/feeds/:site where `:site` is a valid site slug of one of your sites.
* Verify you get redirected to the `/read/feeds/:siteId` route and it works well.
* Go to http://calypso.localhost:3000/read/feeds/asdasdasdsa where `asdasdasdsa` is some random gibberish. 
* Verify you get a "not found" message.
* Verify there is no request to `/read/feeds/NaN` that used to be there in this case.

Note: ~in the last case with the 404 feed, there's an extra network request done to `/read/feeds/NaN`. This occurs in `master` too, and I'm planning to fix that in a subsequent PR.~ I decided to fix this in this PR, as it's a very easy change in the same place where this PR is modifying.
